pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -715,19 +715,16 @@ def _lazy_init_strategy(self) -> None:
 
         from pytorch_lightning.utilities import _IS_INTERACTIVE
 
-        # TODO move is_compatible logic to strategy API
-        interactive_compatible_strategy = (
+        interactive_recomended_strategy = (
```

Comment:
Do you think we should move it to _init_strategy? which fails faster.

Reply:
I was thinking add this to strategy API which could be called at init_strategy. I think in launcher make much more sense. +1

Reply:
Why change the variable from `compatible` to `recommended`?

Reply:
+1, let's keep the previous name

Reply:
> Do you think we should move it to _init_strategy? which fails faster.

We can't because we need to run `configure_launcher` first and this needs to happen in `lazy_init`.


> Why change the variable from compatible to recommended?

I did this because we decoupled the launchers from the strategy. So the concept of being "compatible with interactive environments" has shifted away from the strategy and it made more sense to recomend (spelling error not intended) strategies that come with compatible launchers by default. Anyway, refactored now.